HIVE-27474 Remove unused function resources method param for registerUDF method in Registry

suprithIUB opened the pull request:
<!--
Thanks for sending a pull request!  Here are some tips for you:
  1. If this is your first time, please read our contributor guidelines: https://cwiki.apache.org/confluence/display/Hive/HowToContribute
  2. Ensure that you have created an issue on the Hive project JIRA: https://issues.apache.org/jira/projects/HIVE/summary
  3. Ensure you have added or run the appropriate tests for your PR: 
  4. If the PR is unfinished, add '[WIP]' in your PR title, e.g., '[WIP]HIVE-XXXXX:  Your PR title ...'.
  5. Be sure to keep the PR description updated to reflect all changes.
  6. Please write your PR title to summarize what this PR proposes.
  7. If possible, provide a concise example to reproduce the issue for a faster review.

-->

### What changes were proposed in this pull request?

Remove unused function resources method param for registerUDF method in Registry for cleanup/better readability.  

https://github.com/apache/hive/blob/25892ea409b3351acbd409d1f66f399a43614798/ql/src/java/org/apache/hadoop/hive/ql/exec/Registry.java#L154-L159

Not used by calling code here, likewise there are more places in the same file https://github.com/suprithIUB/hive/blob/14a1f70607db5ae6cf71b6d4343f308a5167581c/ql/src/java/org/apache/hadoop/hive/ql/exec/FunctionRegistry.java#L496-L514








### Why are the changes needed?
Cleanup


### Does this PR introduce _any_ user-facing change?
No

### Is the change a dependency upgrade?
No


### How was this patch tested?
Existing tests

ayushtkn commented:
not sure we can directly remove public methods or change signatures like this, that would be an incompatible change, the way in general is the deprecate the old method & remove it in the next release

suprithIUB commented:
> not sure we can directly remove public methods or change signatures like this, that would be an incompatible change, the way in general is the deprecate the old method & remove it in the next release

Thanks for the quick review, addressed PR feedback, PTAL @ayushtkn

suprithIUB commented:
@ayushtkn Tagging in case you missed my earlier comment.

suprithIUB commented:
> LGTM

@ayushtkn Thanks, I can assume this will be merged soon by one of the committers because we wait for "24 hours have elapsed since the first +1 vote was cast on the Pull Request" as per documentation here https://cwiki.apache.org/confluence/display/Hive/HowToCommit#HowToCommit-Newcommitters. I don't have write access to merge the PR now,  just trying to understand the process.

ayushtkn commented:
Yep, you got that right, I will merge it tmrw
